database/db_test.go

```diff
@@ -4,13 +4,97 @@ import (
 	"testing"
 
 	"github.com/Hukyl/genesis-kma-school-entry/database"
+	"github.com/Hukyl/genesis-kma-school-entry/database/config"
 	"github.com/stretchr/testify/assert"
 )
 
+type MockUser struct {
+	Email    string
+	Password string
+}
+
+type EmptyStruct struct{}
+
 func TestSingletonDBConnection(t *testing.T) {
 	db := database.SetUpTest(t)
 
 	conn1 := db.Connection()
 	conn2 := db.Connection()
 	assert.Equal(t, conn1, conn2)
 }
+
+func TestInvalidConfig(t *testing.T) {
+	config := config.Config{
+		DatabaseService: "unknown",
+		DatabaseDSN:     "unknown",
+	}
+	db, err := database.New(config)
+	assert.Nil(t, db)
+	assert.NotNil(t, err)
+}
+
+func TestInit(t *testing.T) {
+	config := config.Config{
+		DatabaseService: "sqlite",
+		DatabaseDSN:     "file::memory:?cache=shared",
+	}
+	db := database.DB{Config: config}
+	err := db.Init()
+	assert.Nil(t, err)
+}
+
+func TestInitFail(t *testing.T) {
+	config := config.Config{
+		DatabaseService: "unknown",
+		DatabaseDSN:     "file::memory:?cache=shared",
+	}
+	db := database.DB{Config: config}
+	err := db.Init()
+	assert.NotNil(t, err)
+}
+
+func TestMigrateNull(t *testing.T) {
```

Comment:
why don't combine [these](https://github.com/GenesisEducationKyiv/software-engineering-school-4-0-Hukyl/pull/5/files#diff-7f605fc196f60a6f1dd07030d4ba82cf082ef167a8bdd0303b6c615734b34b64R56-R80) 4 tests to one table test?